src/manifest.rs

```diff
@@ -1152,8 +1162,125 @@ struct CargoDeb {
+    #[default]
+    Unknown,
+}
+
+impl MergeByKey {
+    /// Merges w/ a parent asset list
+    /// 
+    fn merge(self, parent: &AssetList) -> AssetList {
+        let merge_map = { 
+            parent.iter().fold(BTreeMap::new(), |parent, asset| {
+                self.prep_parent_item(parent, asset)
+            })
+        };
+
+        self.merge_with(merge_map)
+    }
+
+
+    /// Folds the parent asset into a merge-map preparing to prepare for a merge,
+    /// 
+    /// **Note** No-op if unknown strategy
+    /// 
+    fn prep_parent_item<'a>(&'a self, mut parent: MergeMap<'a>, asset: &'a Vec<String>) -> MergeMap {
+        if let [src, dest, perm, ..] = &asset[..] {
+            match &self {
+                MergeByKey::Src(_) => {
+                    parent.insert(src, [dest, perm]);
+                },
+                MergeByKey::Dest(_) => {
+                    parent.insert(dest, [src, perm]);
+                },
+                MergeByKey::Unknown => {
+                    // No-op
+                    warn!("Unknown merge strategy");
+                },
+            }
+            parent
+        } else {
+            warn!("Incomplete asset entry {:?}", asset);
+            parent
+        }
+    }
+
+    /// Merges w/ a parent merge map and returns the resulting asset list,
+    /// 
+    fn merge_with(&self, parent: MergeMap) -> AssetList {
+        match self {
+            MergeByKey::Src(assets) => {
+               assets.iter().fold(parent, |mut acc, asset| {
+                    if let [src, dest, perm, ..] = &asset[..] {
+                        if let Some([replaced_dest, replaced_perm]) = acc.insert(src, [dest, perm]) {
+                            debug!("Replacing {:?} w/ {:?}", [replaced_dest, replaced_perm], [dest, perm]);
+                        }
+                        acc
+                    } else {
+                        warn!("Incomplete asset entry {:?}", asset);
+                        acc
+                    }
+                }).iter().map(|(src, [dest, perm])| {
+                    vec![src.to_string(), dest.to_string(), perm.to_string()]
+                }).collect()
+            },
+            MergeByKey::Dest(assets) => {
+                assets.iter().fold(parent, |mut acc, asset| {
+                    if let [src, dest, perm, ..] = &asset[..] {
+                        if let Some([replaced_src, replaced_perm]) = acc.insert(dest, [src, perm]) {
+                            debug!("Replacing {:?} w/ {:?}", [replaced_src, replaced_perm], [src, perm]);
+                        }
+                        acc
+                    } else {
+                        warn!("Incomplete asset entry {:?}", asset);
+                        acc
+                    }
+                }).iter().map(|(dest, [src, perm])| {
+                    vec![src.to_string(), dest.to_string(), perm.to_string()]
+                }).collect()
+            },
+            MergeByKey::Unknown => todo!(),
```

Comment:
Should this be an error? Can you remove the unknown key completely?

Reply:
Yep, can definitely remove it.